src/Controller/HomeController.php

```diff
@@ -2,16 +2,38 @@
 
 namespace App\Controller;
 
+use App\Entity\Trick;
+use App\Service\Paginator;
+use Doctrine\Common\Persistence\ObjectManager;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\Routing\Annotation\Route;
 
 class HomeController extends AbstractController
 {
+    public const TRICKS_PER_PAGE = 10;
+
     /**
+     * Show the home page
+     *
      * @Route("/", name="home")
+     * @Route("/home/{page}", name="home_paging", requirements={"page": "\d+"})
```

Comment:
Why not, but you can also handle the pagination without creating a dedicated route:

`
$router->generate('home', ['page' => 10])
// As the route has no `page` parameter, the router will pass this parameter in the query parameters (mywebsite.com/my_route?page=10)

// Controller
$page = $request->query->getInt('page', 1)
`

Reply:
Ok, so I merged the 2 routes into one : `"/{page}"`